### PR TITLE
Addressing site->vpn->route_via issues

### PR DESCRIPTION
**Pull request by infosec-shinobi**
Addressing #198

Aligning these values as "optional" instead of "required" with the documentation.

### CODE REVIEW

**Comment by infosec-shinobi** [comment]
FYI, @dirtyonekanobi  made a good point when chatting with him today that an update to the flatten method for vpn/route_via also needs done. Added a commit for that.

If I get some extra cycles, I will try and throw together some tests that account for default sites.

File: appgate/resource_appgate_site.go [hunk 1]
```diff
@@ -668,8 +668,12 @@ func flattenSiteVPN(currentVersion *version.Version, in openapi.SiteAllOfVpn) []
...
 		routeVia := make(map[string]interface{})
-		routeVia["ipv4"] = in.RouteVia.GetIpv4()
-		routeVia["ipv6"] = in.RouteVia.GetIpv6()
+		if _, o := in.RouteVia.GetIpv4Ok(); o != false {
+			routeVia["ipv4"] = in.RouteVia.GetIpv4()
+		}
+		if _, o := in.RouteVia.GetIpv6Ok(); o != false {
+			routeVia["ipv6"] = in.RouteVia.GetIpv6
+		}
 		m["route_via"] = routeVia
```
Comment (infosec-shinobi) [inline appgate/resource_appgate_site.go:677]: We also have this set to this:
m["route_via"] = []interface{}{routeVia}

I am newish to Go so not sure if that routeVia needs to be an interface or not.

**Comment by infosec-shinobi** [comment]
FYI...

Been mucking with this a bit more. I believe I have it narrowed down to just the "route_via" section under "vpn"... Specifically what was noted in my last comment about 

The error it is throwing (parts removed, including relevant sections):

```
panic: interface conversion: interface {} is map[string]interface {}, not *schema.Set
...
2021-12-27T15:19:34.937Z [DEBUG] plugin.terraform-provider-appgatesdp_v0.8.3: github.com/appgate/terraform-provider-appgatesdp/appgate.resourceAppgateSiteRead()
2021-12-27T15:19:34.937Z [DEBUG] plugin.terraform-provider-appgatesdp_v0.8.3: 	github.com/appgate/terraform-provider-appgatesdp/appgate/resource_appgate_site.go:601
...
2021-12-27T15:19:34.941Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/appgate/appgatesdp/0.8.3/linux_amd64/terraform-provider-appgatesdp_v0.8.3 pid=320 error="exit status 2"
2021-12-27T15:19:34.941Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021/12/27 15:19:34 [DEBUG] appgatesdp_site.dv_test_site: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2021/12/27 15:19:34 [TRACE] eval: *terraform.EvalMaybeTainted
2021/12/27 15:19:34 [TRACE] EvalMaybeTainted: appgatesdp_site.dv_test_site encountered an error during creation, so it is now marked as tainted
...
```

The interesting thing is... this actually does create the site... But TF seems to not be able to read/validate. RPC errors occurring on both creating a new site and when trying to run a plan against a state that has pre-existing sites that have the "route_via" section included under "vpn".

If you want to test, you can use this sample resource:

```
resource "appgatesdp_site" "default_test_site" {
    name = "default test"
    short_name = "DT"
    entitlement_based_routing = false
    notes = "Managed by terraform"
    default_gateway {
        enabled_v4       = true
        enabled_v6       = false
    }
    name_resolution {
        use_hosts_file = false
        dns_resolvers {
            name = "test"
            update_interval = 60
            servers = ["10.10.10.10"]
            search_domains = ["example.com"]
        }
    }
    vpn {
        state_sharing = false
        ip_access_log_interval_seconds = 120
        snat = false
        tls {
          enabled = true
        }
        dtls {
            enabled = false
        }
        route_via {
            ipv4 = "10.10.10.10"
            ipv6 = "fdf8:f53b:82e4::53"
        }
      }
    tags = ["terraform", "api-created","default_test_site"]
}
```

Added another pr which I think addresses the panic... I don't think route_via was being returned as the correct type.

File: appgate/resource_appgate_site.go [hunk 2]
```diff
@@ -668,9 +668,13 @@ func flattenSiteVPN(currentVersion *version.Version, in openapi.SiteAllOfVpn) []
...
 	if in.HasRouteVia() && in.RouteVia.Ipv4 != nil {
 		routeVia := make(map[string]interface{})
-		routeVia["ipv4"] = in.RouteVia.GetIpv4()
-		routeVia["ipv6"] = in.RouteVia.GetIpv6()
-		m["route_via"] = routeVia
+		if _, o := in.RouteVia.GetIpv4Ok(); o != false {
+			routeVia["ipv4"] = in.RouteVia.GetIpv4()
+		}
+		if _, o := in.RouteVia.GetIpv6Ok(); o != false {
+			routeVia["ipv6"] = in.RouteVia.GetIpv6
```
Comment (danijeel) [inline appgate/resource_appgate_site.go:675]:
```suggestion
			routeVia["ipv6"] = in.RouteVia.GetIpv6()
```

:) we just missed to call the function here `()`

**Review by danijeel — APPROVED** [review]
Great work finding this bug! Thank you for your contribution, I will create a new release later today.

**Pull request merged**